atlasdb-commons/src/test/java/com/palantir/async/initializer/CallbackTest.java

```diff
@@ -40,6 +46,28 @@ public void cleanupExceptionGetsPropagatedAndStopsRetrying() {
         assertThat(countingCallback.initCounter).isEqualTo(5L);
     }
 
+    @Test
+    public void shutdownDoesNotBlockWhenTaskIsNotRunning() {
+        CountingCallback countingCallback = new CountingCallback(true);
+
+        long start = System.currentTimeMillis();
+        countingCallback.blockUntilSafeToShutdown();
+        assertThat(System.currentTimeMillis()).isLessThanOrEqualTo(start + 100L);
```

Comment:
Flagging that this seems a little risky - is 0.1 seconds a big enough window for a slammed CircleCI node?

Not blocking on this flag, as it could be fine in practice, and I guess there's always FlakeRule.

Reply:
Should be fine since it should really not block at all, but if it flakes we can increase it